tests/countries/test_turks_and_caicos_islands.py

```diff
@@ -0,0 +1,219 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.turks_and_caicos_islands import TurksAndCaicosIslands, TC, TCA
+from tests.common import CommonCountryTests
+
+
+class TestTC(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(TurksAndCaicosIslands, years=range(2010, 2025))
+
+    def test_country_aliases(self):
+        self.assertAliases(TurksAndCaicosIslands, TC, TCA)
+
+    def test_commonwealth_day(self):
+        name = "Commonwealth Day"
+        self.assertHolidayName(
+            name,
+            "2020-03-09",
+            "2021-03-08",
+            "2022-03-14",
+            "2023-03-13",
+            "2024-03-11",
+            "2025-03-10",
+        )
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        self.assertHolidayName(
+            name,
+            "2020-04-10",
+            "2021-04-02",
+            "2022-04-15",
+            "2023-04-07",
+            "2024-03-29",
+            "2025-04-18",
+        )
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+        self.assertHolidayName(
+            name,
+            "2020-04-13",
+            "2021-04-05",
+            "2022-04-18",
+            "2023-04-10",
+            "2024-04-01",
+            "2025-04-21",
+        )
+
+    def test_jags_mccartney_day(self):
+        name_pre_2020 = "National Heroes Day"
+        name_2020 = "JAGS McCartney Day"
+
+        self.assertHolidayName(
+            name_2020,
+            "2020-05-25",
+            "2021-05-31",
+            "2022-05-30",
+            "2023-05-29",
+            "2024-05-27",
+            "2025-05-26",
+        )
+
+        self.assertHolidayName(name_pre_2020, range(2010, 2020))
+        self.assertHolidayName(name_2020, range(2020, 2026))
+        self.assertNoHolidayName(name_pre_2020, range(2020, 2026))
+        self.assertNoHolidayName(name_2020, range(2010, 2020))
```

Comment:
```suggestion
        self.assertHolidayName(name_pre_2020, range(1963, 2020))
        self.assertHolidayName(name_2020, range(2020, 2050))
        self.assertNoHolidayName(name_pre_2020, range(2020, 2050))
        self.assertNoHolidayName(name_2020, range(1963, 2020))
```
And the same for rest of holidays with similar name changes.